sat/apiclient/JobstatStateChecker.py

```diff
@@ -0,0 +1,60 @@
+#
+# MIT License
+#
+# (C) Copyright 2022 Hewlett Packard Enterprise Development LP
+#
+# Permission is hereby granted, free of charge, to any person obtaining a
+# copy of this software and associated documentation files (the "Software"),
+# to deal in the Software without restriction, including without limitation
+# the rights to use, copy, modify, merge, publish, distribute, sublicense,
+# and/or sell copies of the Software, and to permit persons to whom the
+# Software is furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included
+# in all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL
+# THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR
+# OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+# ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+# OTHER DEALINGS IN THE SOFTWARE.
+#
+"""
+Client for querying the Job State Checker service
+"""
+
+import logging
+
+from sat.apiclient.gateway import APIError, APIGatewayClient
+
+LOGGER = logging.getLogger(__name__)
+
+
+class JobstatStateChecker(APIGatewayClient):
+    base_resource_path = 'statechecker/jobstat/'
+
+    def get_all(self):
+        """Get all results from Job State Checker.
+
+        Returns:
+            A list of dictionaries where each dictionary pertains to a
+            single job.
+
+        Raises:
+            APIError: if there is a failure querying the Job State Checker API
+                or getting the required information from the response.
+        """
+        err_prefix = 'Failed to get State Checker data'
+        try:
+            response = self.get('all').json()
+            if response['errorcode'] != 0:
+                raise APIError(response['errormessage'])
```

Comment:
This is pretty close to what I was hoping for, thanks for addressing all the feedback here.

I see you mentioned that the response objects always have `errorcode` and `errormessage` keys (even in the successful case) but I think it would be good to be more defensive and handle when they aren't there, i.e. first check that the expected keys exist.

Otherwise, in the case when there is successful `jobstat` data but there is no errorcode, we will fail with an `APIError` (due to missing key). Other (edge cases) thoughts I have are how we are handling if `errorcode` is a null type, and how we are handling if the 'errormessage' key is missing.

Just in case the behavior of the API changes, or something unexpected happens.

How would you feel about this approach, I think it's a bit more defensive:
```suggestion
            if 'errorcode' in response and response['errorcode']:
                raise APIError(response.get('errormessage', 'unknown error'))
```